src/cockpit/channels/http.py

```diff
@@ -114,7 +114,7 @@ def request(self):
                 command='response', status=response.status, reason=response.reason,
                 headers=self.parse_headers(response.headers)))
             self.read_send_response(response)
-        except http.client.HTTPException as error:
+        except (http.client.HTTPException, ConnectionError) as error:
```

Comment:
Why not `OSError` more generally?  This gets raised as a top-level task crash otherwise, right?  Let's try to be more robust here.

Reply:
Sure! Done.